cmd/algod/main.go

```diff
@@ -282,6 +256,43 @@ func main() {
 	}
 	fmt.Fprintf(os.Stdout, "Deadlock detection is set to: %s (Default state is '%s')\n", deadlockState, config.DefaultDeadlock)
 
+	if log.GetTelemetryEnabled() {
+		// Make a copy of config and reset DNSBootstrapID in case it was disabled.
+		cfgCopy := cfg
+		cfgCopy.DNSBootstrapID = telemetryDNSBootstrapID
+
+		// Periodically check SRV records for new telemetry URI
+		go srvUpdaterLoop(1*time.Minute, cfgCopy, s.Genesis.Network, log)
+
+		currentVersion := config.GetCurrentVersion()
+		startupDetails := telemetryspec.StartupEventDetails{
+			Version:      currentVersion.String(),
+			CommitHash:   currentVersion.CommitHash,
+			Branch:       currentVersion.Branch,
+			Channel:      currentVersion.Channel,
+			InstanceHash: crypto.Hash([]byte(absolutePath)).String(),
+		}
+
+		log.EventWithDetails(telemetryspec.ApplicationState, telemetryspec.StartupEvent, startupDetails)
+
+		// Send a heartbeat event every 10 minutes as a sign of life
+		ticker := time.NewTicker(10 * time.Minute)
+		go func() {
```

Comment:
This is another place where we need to shut down correctly. It's ok if you intend to do that in a subsequent change.